### ISSUE TITLE
Problem with DialogFlow v2 integration 

### ISSUE THREAD

**Issue by atsoares** [botman/botman#1083]
- BotMan Version: 2.5.0
- PHP Version: 7.1.8

### Description:

Hi, I'm trying to integrate DialogFlow with botman to be able to use botman widget, but the documentation steps doesn't work. Don't know if I'm passing the right token, I created a new service-account-file-key but it doesn't work. The fact is Dialogflow v2 doesn't use token access anymore, it uses OAuth 2 for Server-Server Applications, so its a different way to connect.

here is my chat.php file: 

```
<?php
   require_once 'vendor/autoload.php';

   use BotMan\BotMan\BotMan;
   use BotMan\BotMan\BotManFactory;
   use BotMan\BotMan\Drivers\DriverManager;
   use BotMan\BotMan\Middleware\Dialogflow;
   
   $config = ['web'=>['matchingData'=>['driver'=>'web']]];

   DriverManager::loadDriver(\BotMan\Drivers\Web\WebDriver::class);
   $botman = BotManFactory::create($config);

   $dialogflow = Dialogflow::create('DIALOGFLOW-TOKEN')->listenForAction();
   $botman->middleware->received($dialogflow);

  $botman->hears('input.welcome', function(Botman $bot) {
	$extras = $bot->getMessage()->getExtras();
	$apiReply = $extras['apiReply'];
	$apiAction = $extras['apiAction'];
	$apiIntent = $extras['apiIntent'];

	$bot->reply($apiReply);
  })->middleware($dialogFlow);

  // Start listening
  $botman->listen();

?>
```

Please can someone make a better documentation for [nlp integration]( https://botman.io/2.0/nlp ) or give me some help??

**Comment by somato-tomato** [comment]
I think botman v2 still not support Dialogflow API V2

**Comment by ranrinc** [comment]
@atsoares please do check out pull request.. Someone from the community create a new DialogFlowV2.php which support it. He have been waiting for someone to assist him. I'm also pulling his changes to include on my current botman project. https://github.com/botman/botman/pull/1010

Hope this help

**Comment by atsoares** [comment]
@ranrinc thanks for that, but actually Dialogflow V1 API shutdown has been postponed to March 31st, 2020, so until them I will keep using the normal version. 

The problem is the bot doesn't reply when I use ` hears('input.welcome'` and I don't know what is wrong. I've checked already at Dialogflow's log and looks like all the messages that I type from the web widget the Dialogflow receive, but I the bot doesn't reply. 

I tried with basic hearing documentation and works but using hears with middleware don't.
